### ISSUE TITLE
REGRESSION: default resolvers

### ISSUE THREAD

**Issue by rafales** [graphql-python/graphene-sqlalchemy#234]
This commit: https://github.com/graphql-python/graphene-sqlalchemy/commit/9a0f7400957c5b5f5f8e952db2502ef1f79fac1f

started setting resolver on fields. Providing a resolver for all fields breaks usage of `default_resolver` (meta param to ObjectType), which our project was using. This introduced a silent error, where our default resolver was not called.

**Comment by jnak** [comment]
Hello, Thanks for reporting the issue. Can you share more about how you relied on the `default_resolver`? I believe `graphene-sqlalchemy` had a big dependency on `attr_resolver` so it's not clear to me how it could work with a different `default_resolver`.

**Comment by rafales** [comment]
Up until the latest version (2.2.1) graphene-sqlalchemy did not use any custom resolvers.

We have something built on top of graphene that uses `default_resolver` for some access-related logic. Basically what happens is something like this:

```python
attrs_dict = {
                "Meta": {
                    "model": self.db_model,
                    "default_resolver": self.default_field_resolver,
                    "only_fields": only_fields or (),
                    "exclude_fields": exclude_fields or (),
                }
            }

            graphene_type = type(
                self.db_model.__name__, (SQLAlchemyObjectType,), attrs_dict
            )
```

**Comment by rafales** [comment]
Everything is fine and our `default_field_resolver` gets called in versions `<2.2.1`, but in `2.2.1` `default_resolver` never gets called for `SQLAlchemyObjectType` descendants.

**Comment by jnak** [comment]
Would you mind explaining what type of logic you are putting in `default_field_resolver`? We need understand the use case if we are going to consider supporting explicitly it via test and documentation.

**Comment by dtamez** [comment]
In our case we had a few places where we had a resolver on a type but did not explicitly have the field on the type.  The type was defined as a SQLAlchemyObjectType and the field(s) in question were not in the excluded fields list.  Before 2.2.1 the resolvers for those fields were being called and executed.  With 2.2.1 the resolvers no longer got called.  For now we have pinned the lib to 2.2.0 until this gets fixed or a decision is made that it should not be fixed.

**Comment by davidmoss** [comment]
I experienced this regression too, where my `resolver_<field>` stopped getting called in v2.2.1 for objects in a queryset.

**Comment by jnak** [comment]
Can anyone share an actual repro of the bug? thank you

**Comment by ticosax** [comment]
@jnak please check #235 to repro the bug, if I understand correctly

**Comment by davidmoss** [comment]
@ticosax looks good to me

**Comment by jnak** [comment]
Apologies for the late reply. I was on vacation and did not have time to look into this until this week.

Thanks @ticosax for the repro. I figured out what's going on. It's indeed a serious regression :/ 
I have a fix that is mostly done, I just need to address a TODO. I'll finish it early next week.
Here is the [branch](https://github.com/jnak/graphene-sqlalchemy/commits/default-resolver) in case you need the fix now.

Cheers

**Comment by jnak** [comment]
@davidmoss @ticosax @dtamez Here is the fix: https://github.com/graphql-python/graphene-sqlalchemy/pull/241

@rafales Does my PR fix your issue? It seems that the repro posted by @ticosax is different from your original issue with `default_resolver`. I don't understand how you are using `default_resolver` without interfering with `SQLAlchemyObjectType` resolve its fields. Can you share some more details?

**Comment by jnak** [comment]
The fix has been released on [2.2.2](https://github.com/graphql-python/graphene-sqlalchemy/releases/tag/2.2.2 ). 

@dtamez Closing for now since you haven't provided yet more detail. Feel free to re-open when you have more details to share.
